### DISCUSSION TITLE
Ability to configure context menu only when required

### DISCUSSION THREAD

**Discussion by sha2nktiwade** [tinymce/tinymce#8857] [General]
**What is the current behavior? Describe the bug**
1- When there is no text in the editor, do not want the editor to show context menu options.
2- Want the context menu to appear and show options only for the plugins I have configured in.

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**
1- If contextmenu: false is set to false, you will not get any context menu options even when you have configured it for some specific plugins with  editor.ui.registry.addContextMenu('conditionalsection', 
2- It should show the context menu only for the plugins in which we have configured it in.

**What is the expected behavior?**
User should have the option to not configure any context menu options by default, only when they are inside a specific plugin and want to configure any options separately they should able to do that.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
Vresion we are using is TinyMCE 6.1.3, not dependent on browser.

**Comment by lostkeys** [comment]
Hi @sha2nktiwade. Use the `contextmenu` but only add your custom context menu to it, omitting all the default ones. Try that and report back if it solved your problem.

**Comment by sha2nktiwade** [comment]
Hi @lostkeys - Thank you for response. I did configure the custom plugins I needed in the contextmenu like below:
 contextmenu: 'articlelink conditionalsection ',
So, when I right-click now in the empty editor, I get these two plugins.

Additionally, when I have configured data with plugin 'conditionalsection' I only want to get options like 'Edit Conditional Section' an 'Delete Conditional Section'. I am replacing the 'conditionalsection' with these two options. But I am also getting 'articlelink' here like below:
![image](https://github.com/tinymce/tinymce/assets/130130998/70de9dbf-4a6f-4af8-b166-8f5bb6d52cf9)


How to get rid of the 'Article Link' option from here?

**Comment by sha2nktiwade** [comment]
in Addition to above, I was checking table and link plugin, and configured both of them in contextmenu inside tinymce.init() method like below:
contextmenu: 'table link articlelink conditionalsection ',


now if I right-click inside the empty editor, I get below options:
![image](https://github.com/tinymce/tinymce/assets/130130998/381b6459-f079-46e9-8546-d2de9db5988e)

If I right-click inside table I get below:
![image](https://github.com/tinymce/tinymce/assets/130130998/02730e89-acf0-4a3b-a2cb-ad1db452966a)

I want similar thing for my custom plugins. Whenever I click inside the plugin, then I should get option only specific to that plugin and remove the other unwanted options. How can I do this?

**Comment by lostkeys** [comment]
Have a look at this [docs page on how to selectively hide and show context menu items](https://www.tiny.cloud/docs/tinymce/6/contextmenu/#defining-a-context-menu-section). The example show how to show an image context menu, only when right clicking on an image. So with this code snippet you should be able to piece together what you need.
